### PR TITLE
HikariCP config fix

### PR DESCRIPTION
**Pull request by saquino0827**
# Add a PR title

Fixes `maxLifetime` settings for HikariCP.

## Issue

Add a link to the issue here.  Consider using
[closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
if the this PR isn't for a story (stories will be closed through different means).

## Checklist

- [ ] I have added tests to cover my changes
- [ ] I have added logging where useful (with appropriate log level)
- [ ] I have added JavaDocs where required
- [ ] I have updated the documentation accordingly

**Note**: You may remove items that are not applicable

### CODE REVIEW

File: shared/src/main/java/gov/hhs/cdc/trustedintermediary/external/HikariConnectionPool.java [hunk 1]
```diff
@@ -41,7 +43,8 @@ static HikariConfig constructHikariConfig() {
         String serverName = ApplicationContext.getProperty("DB_URL", "");
         String dbName = ApplicationContext.getProperty("DB_NAME", "");
         String dbPort = ApplicationContext.getProperty("DB_PORT", "");
-        String connectionLifetime = ApplicationContext.getProperty("DB_MAX_LIFETIME", "1800000");
+        long connectionLifetime =
+                parseLong(ApplicationContext.getProperty("DB_MAX_LIFETIME", "1800000"));
```
Comment (halprin) [inline shared/src/main/java/gov/hhs/cdc/trustedintermediary/external/HikariConnectionPool.java:47]: Let's not static import a method from a class.

```suggestion
                Long.parseLong(ApplicationContext.getProperty("DB_MAX_LIFETIME", "1800000"));
```

Comment (saquino0827) [reply shared/src/main/java/gov/hhs/cdc/trustedintermediary/external/HikariConnectionPool.java]: Fixed.

**Review by halprin — COMMENTED** [review]
Nice, but I would address the CodeQL notice.

File: shared/src/main/java/gov/hhs/cdc/trustedintermediary/external/HikariConnectionPool.java [hunk 2]
```diff
@@ -41,7 +41,8 @@
         String serverName = ApplicationContext.getProperty("DB_URL", "");
         String dbName = ApplicationContext.getProperty("DB_NAME", "");
         String dbPort = ApplicationContext.getProperty("DB_PORT", "");
-        String connectionLifetime = ApplicationContext.getProperty("DB_MAX_LIFETIME", "1800000");
+        long connectionLifetime =
+                Long.parseLong(ApplicationContext.getProperty("DB_MAX_LIFETIME", "1800000"));
```
[the remark this answers is not in the document]
Comment (halprin) [reply shared/src/main/java/gov/hhs/cdc/trustedintermediary/external/HikariConnectionPool.java:45]: If `NumberFormatException` is thrown, maybe just default to `1800000`?

**Review by halprin — APPROVED** [review]

**Comment by saquino0827** [comment]
> Nice, but I would address the CodeQL notice.

Fixed that and added some test coverage with @luis-pabon-tf

**Pull request merged**